src/services/car/__tests__/leasing.test.ts

```diff
@@ -22,4 +22,117 @@ describe("CAR service", () => {
       ).toEqual(null)
     })
   })
+
+  describe("sendLeasingInterest", () => {
+    const leasingInterest = (attributes = {}) => ({
+      name: "Test",
+      phone: "+41781234567",
+      email: "test@test.com",
+      message: "This is a message of a interested customer",
+      ...attributes,
+    })
+
+    describe("valid parameters", () => {
+      beforeEach(() => {
+        fetchMock.mockResponse("")
+      })
+
+      describe("validate only", () => {
+        it("calls validation endpoint", async () => {
+          await sendLeasingInterest(12345, leasingInterest(), {
+            validateOnly: true,
+          })
+
+          expect(fetch).toHaveBeenCalledWith(
+            expect.stringMatching(
+              /\/listings\/12345\/leasing\/interests\/validate$/
+            ),
+            expect.any(Object)
+          )
+        })
+
+        it("returns a success", async () => {
+          const result = await sendLeasingInterest(12345, leasingInterest(), {
+            validateOnly: true,
+          })
+
+          expect(result).toEqual({ tag: "success", result: leasingInterest() })
+        })
+      })
+
+      describe("submit", () => {
+        it("calls submission endpoint", async () => {
+          await sendLeasingInterest(12345, leasingInterest())
+
+          expect(fetch).toHaveBeenCalledWith(
+            expect.stringMatching(/\/listings\/12345\/leasing\/interests$/),
+            expect.any(Object)
+          )
+        })
+
+        it("sends recaptcha token in a header", async () => {
+          await sendLeasingInterest(12345, leasingInterest(), {
+            recaptchaToken: "token",
+          })
+
+          expect(fetch).toHaveBeenCalledWith(expect.any(String), {
+            headers: expect.objectContaining({
+              "Recaptcha-Token": "token",
+            }),
+            body: expect.any(String),
+            method: "POST",
+          })
+        })
+
+        it("returns a success", async () => {
+          const result = await sendLeasingInterest(12345, leasingInterest())
+
+          expect(result).toEqual({ tag: "success", result: leasingInterest() })
+        })
+      })
+    })
+
+    describe("invalid parameters", () => {
+      const errors = [{ param: "email", error: "validations.invalid-format" }]
+
+      beforeEach(() => {
+        fetchMock.mockResponse(
+          JSON.stringify({ errors, message: "validations.not-valid" }),
+          {
+            status: 400,
+          }
+        )
+      })
+
+      it("returns a failure", async () => {
+        const result = await sendLeasingInterest(
+          12345,
+          leasingInterest({ email: "test@test" })
+        )
+
+        expect(result.tag).toEqual("error")
+        if (result.tag === "error") {
+          expect(result.errors).toEqual(errors)
```

Comment:
Isn't this testing weather the mock gets applied to the mock response?

Reply:
It's testing that the error is properly handled, no?